source/unified-test-format/unified-test-format.rst

```diff
@@ -2363,6 +2417,35 @@ An example of this operation follows::
         connections: 1
 
 
+Special Placeholder Value
+-------------------------
+
+$$placeholder
+~~~~~~~~~~~~~
+
+Syntax::
+
+  { field: { $$placeholder: {} } }
+
+This reserved key-value pair can be used anywhere the value for a key might be
+specified in an test file. It is intended to act as a placeholder value in
+contexts where the test runner cannot provide a definite value or may be
+expected to replace the placeholder with a value that cannot be specified by the
+test file (e.g. KMS provider credentials). The test runner MUST raise an error
+if a placeholder value is used in an unexpected context.
+
+An example of using this placeholder value follows::
+
+    kmsProviders:
+      aws:
+        accessKeyId: { $$placeholder: {} }
+        privateAccessKey: { $$placeholder: {} }
+
+Note that ``field: { $$placeholder: {} }`` is not equivalent to ``field: {}``.
+The former indicates ``field`` has an unspecified value, whereas the latter
+indicates ``field`` has the value ``{}`` (an empty document).
```

Comment:
Avoiding an empty document in the example in favor of `{ $$placeholder: 1 }` might also allow you to remove this entire paragraph.